prog/modes/modes.F90

```diff
@@ -29,6 +34,18 @@ program modes
   real(dp), allocatable :: displ(:,:,:)
 
   character(lc) :: lcTmp, lcTmp2
+  integer :: fdUnit
+
+#:if WITH_MPI
+  !> MPI environment, if compiled with mpifort
+  type(TMpiEnv) :: mpiEnv
+
+  ! As this is serial code, trap for run time execution on more than 1 processor with an mpi enabled
+  ! build
+  call mpifx_init_thread(requiredThreading=MPI_THREAD_FUNNELED)
```

Comment:
Any reason to only have this for modes?

Reply:
Probably should make this a function in a module that can be imported by programs in the main build system (there are some Fortran snippets that aren't fully integrated inside prog/transporttools though).

I think the eternal module should be a separate PR though, so will just merge.